pod/perldelta.pod

```diff
@@ -66,6 +66,17 @@ C<INCDIR> methods will no longer trigger an exception, and instead will
 be treated the same as unblessed coderefs are, and executed as though
 they were an C<INC> hook.
 
+=head2 Forbidden control flow out of C<defer> or C<finally> now detected at compile-time
+
+It is forbidden to attempt to leave a C<defer> or C<finally> block by means
+of control flow such as C<return> or C<goto>. Previous versions of perl could
+only detect this when actually attempted at runtime.
+
+This version of perl adds compile-time detection for many cases that can be
+statically determined. This may mean that code which previously compiled
+successfully is now reported as an error, but only in cases where it would in
```

Comment:
> but only in cases where it would in any case have been an error to actually execute the code anyway.

I re-read this sentence many times to understand it.
Perhaps, we can supply an example or leave just the main point?

Reply:
I'll see if I can reword it somehow. Maybe:

"This version of perl adds compile-time detection for many cases that can be statically determined. This may mean that code which compiled successfully on a previous version of perl is now reported as a compile-time error with this one. This only happens in cases where it would in any case have been an error to actually execute the code anyway."

Reply:
> in cases where it would in any case have been an error to actually execute the code anyway.

The **clause** still remains unchanged :-) 

The new compile-time out-of-block error only happens where it was already an error?

What's the meaning of this, why do we need to mention that, or maybe it's just me not getting the point yet (sorry, then) :-)

Reply:
I think we can say something like: out-of-block was already an error in previous Perl versions. But the error is now determined on compile-time rather than runtime as previously hence improving the speed of execution (if I got that right).

Reply:
It's no faster to execute as we still have to detect those conditions anyway, because we can't *always* detect it statically. It just means that some kinds of programs that were always going to be an error are determined to be so at compiletime, rather than waiting until runtime to find out.

Reply:
I've rewritten the words in perldelta now.

The user-facing reason is "because compiletime is better than runtime".

Additionally I wanted to use this new function in the feature-class branch, for protecting the expressions in `field $var = EXPR` so I needed the code around anyway. Since it's here (because I copied it from `Syntax::Keyword::Defer`) we might as well use it for `defer` and `finally` blocks too.